lib/_http_outgoing.js

```diff
@@ -109,81 +111,187 @@ OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function(error) {
-  if (this.socket)
-    this.socket.destroy(error);
-  else
-    this.once('socket', function(socket) {
+  const socket = this.socket;
```

Comment:
The only benefit that I thought about is character count :) There are at least four occurrences of `socket` down there.